src/shared/components/modal/modal-utils.ts

```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
+import { ModalContext } from './ModalProvider';
+
+export const useModalLauncher = () => {
+  const showModal = React.useContext(ModalContext);
+
+  return showModal;
+};
```

Comment:
you can move this to ModalProvider.tsx itself
```suggestion
export const useModalLauncher = () => React.useContext(ModalContext);
```